File: bots/Bot_jQuery.py
#!/usr/bin/python
# vi:si:et:sw=4:sts=4:ts=4
# -*- coding: UTF-8 -*-
# -*- Mode: Python -*-

from lib.AppdateHTTPRetriever import *

class Bot_jQuery(AppdateHTTPRetriever):
  """jQuery BOT"""
  
  def identify(self):
    return {
      'projectGroup': 'jQuery',
      'projectName': 'jQuery',
      'projectDescription': 'The jQuery JavaScript library',
      'projectWebSite': 'http://jquery.com'
    }
  
  def run(self):
    
    ##
    # Current Version
    ##
    self.sniffer.setBaseUrl('http://jquery.com')
    self.sniffer.setTargetPath('/')
    data = self.sniffer.setTargetPattern("""<p class="jq-version"><strong>Current Release:</strong>(.*)</p>""").run()
  
    if not data:
      self.say('Unable to retrieve version')
      return KO
      
    self.set('currentVersion', data.group(1).strip())
    
    ##
    # Release Notes is unknown
    ##
    self.set('releaseNotesUrl', "")
    
    ##
    # Release Date is unknown
    ##
    self.set('releaseDate', "")
  
    ##
    # Download URL
    ##
    self.set('downloadUrl', 'http://code.google.com/p/jqueryjs/downloads/list');
    #data = self.sniffer.setTargetPattern("""<form action="(http\:\/\/code.google.com\/p\/jqueryjs\/downloads\/detail)" method="get">""").run()
    #if data is not None:
    #  self.set('downloadUrl', data.group(1).strip())

    return OK

if __name__ == "__main__":
  bot = Bot_jQuery()
  if bot.isRunnable():
    rc = bot.run()
    if rc is not OK:
      print "An error occurred while running this bot"
    else:
      print bot.getProperties()
      print bot.getOutput()

File: bots/js/Bot_jQuery.js
var Bot = require('appdate-bot').Bot,
    sprintf = require('util').format
;

var bot = new Bot();

bot.open('https://github.com/jquery/jquery/releases')

    .then(function (response) {

        var $tags = response.$('.release-timeline-tags > li');

        var $r1 = $tags.eq(0);
        var $r2 = $tags.eq(1);

        bot.set('releaseDate', $r1.find('time').attr('datetime'));
        bot.set('currentVersion', $r1.find('h3 > a > .tag-name').text());
        bot.set('downloadUrl', sprintf('http://code.jquery.com/jquery-%s.js', bot.get('currentVersion')));
        bot.set('downloadPage', 'http://jquery.com/download/');

        /* Version 2 */
        // console.log($r2.find('time').attr('datetime'));
        // console.log($r2.find('h3 > a > .tag-name').text());
        // console.log($r2.find('h3 > a').attr('href'));

        return bot.open('http://jquery.com/download/');
    })

    .then(function (response) {

        var $rn1 = response.$(sprintf('a:contains("jQuery %s release notes")', bot.get('currentVersion')));

        bot.set('releaseNotesUrl', $rn1.attr('href'));
    })

    .then(function () {
        console.log(bot.results);
    })

    .catch(function (err) {
        console.log(err);
    });
